cmake/GenerateVersion.cmake

```diff
@@ -24,7 +27,7 @@ if(NVIM_VERSION_PRERELEASE)
   set(NVIM_VERSION "${NVIM_VERSION}-${NVIM_VERSION_GIT}")
 endif()
 
-set(NVIM_VERSION_STRING "#define NVIM_VERSION_MEDIUM \"${NVIM_VERSION}\"\n")
+set(NVIM_VERSION_STRING "#define NVIM_VERSION_MEDIUM \"${NVIM_VERSION}\"\n#define NVIM_VERSION_BUILD \"${NVIM_VERSION_BUILD}\"\n")
```

Comment:
`CURRENT_VERSION_HASH` is based on NVIM_VERSION_STRING, the comparison will always fail:

    if(NOT "${NVIM_VERSION_HASH}" STREQUAL "${CURRENT_VERSION_HASH}")

Reply:
The `NVIM_VERSION_STRING` is what is written into the `OUTPUT` file.

The `SHA1` of the `OUTPUTFILE` will be different from `NVIM_VERSION_STRING` on the initial build on a new git state, but they'll be the same on subsequent rebuilds.

The comparison works correctly on my machine (look for `Updating Neovim version file contents: ...` message:

https://github.com/neovim/neovim/assets/58627896/616e198f-0285-4985-8a58-60d7e18695f9